client/js/components/procedure/admin/InstitutionTagManagement/InstitutionList.vue

```diff
@@ -726,6 +743,55 @@ export default {
       this.currentlySelectedFilterCategories = selectedCategories
     },
 
+    setFilterOptionsFromFilterQuery () {
+      const filterQueryFromStorage = this.getFilterQueryFromLocalStorage()
+      const categoryIdsWithSelectedFilterOptions = Object.keys(filterQueryFromStorage)
+        .filter(id => id.includes('_group'))
+        .map(id => id.replace('_group', ''))
+
+      categoryIdsWithSelectedFilterOptions.forEach(id => {
+        const selectedFilterOptionIds = Object.values(filterQueryFromStorage)
+          .filter(el => el.condition?.memberOf === `${id}_group`)
+          .map(el => el.condition.value)
+
+        this.setInitialFlyoutFilterIds({ categoryId: id, filterIds: selectedFilterOptionIds })
+      })
+    },
+
+    /**
+     * Sets appliedFilterQuery if a filterQuery is stored in localStorage
+     * appliedFilterQuery is then used to create querIds
+     * queryIds are passed as initialQueryIds to FilterFlyout
+     * on mounted of the FilterFlyout, the filterOptions are then requested and created
+     * The selected filter option ids are written to the store
+     * the watcher in FilterFlyout is triggered and updates the currentQuery
+     * with the selected filter option ids from localStorage
+     */
+    setAppliedFilterQueryFromStorage () {
+      const filterQueryFromStorage = this.getFilterQueryFromLocalStorage()
+
+      this.setAppliedFilterQuery(filterQueryFromStorage)
+    },
+
+    setFilterQueryFromStorage () {
+      const filterQueryFromStorage = this.getFilterQueryFromLocalStorage()
+      const filterIds = Object.keys(filterQueryFromStorage)
+
+      if (filterIds.length > 0) {
```

Comment:
I am pretty sure, that this check is unnecessary, since even with a length of `0` it just would itterate zero times,